x/bep3/keeper/swap.go

```diff
@@ -233,11 +236,21 @@ func (k Keeper) UpdateExpiredAtomicSwaps(ctx sdk.Context) error {
 
 	// Expire incomplete swaps (claimed swaps have already been removed from byBlock index)
 	for _, id := range expiredSwaps {
-		swap, _ := k.GetAtomicSwap(ctx, id)
-		swap.Status = types.Expired
-		k.SetAtomicSwap(ctx, swap)
-		k.RemoveFromByBlockIndex(ctx, swap)
+		atomicSwap, _ := k.GetAtomicSwap(ctx, id)
+		atomicSwap.Status = types.Expired
+		k.SetAtomicSwap(ctx, atomicSwap)
+		k.RemoveFromByBlockIndex(ctx, atomicSwap)
+
+		// Emit 'swap_expired' event
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeSwapExpired,
+				sdk.NewAttribute(types.AttributeKeyAtomicSwapID, fmt.Sprintf("%s", hex.EncodeToString(atomicSwap.GetSwapID()))),
+				sdk.NewAttribute(types.AttributeExpirationBlock, fmt.Sprintf("%d", ctx.BlockHeight())),
+			),
+		)
```

Comment:
you can add all the swapIDs to an attribute array and then emit a single event

Reply:
https://github.com/Kava-Labs/kava/pull/450/commits/94cc38c3d22bae9a6178174be46bc0ef128797d9